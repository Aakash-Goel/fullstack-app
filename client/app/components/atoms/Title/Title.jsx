/**
 * Themed Title
 */

import React from 'react';
import classnames from 'classnames';
import { object, oneOf, node, string } from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';

import titleStyles from './Title.style';

/**
 * @param {object} [classes] - The style definition available here.
 * @param {number} level - The heading level. (i.e., The numbers of h1, h2, h3, h4, h5, h6) [ 1 | 2 | 3 | 4 | 5 | 6 ]
 * @param {object} [img=null] - The definition of the optional image. Defaults to null.
 * @param {node} [children=null] - The children of the component. Defaults to null.
 * @param {string} [className] - CSS class name to allow custom styling to be passed in. Defaults to empty string.
 */
const propTypes = {
  classes: object.isRequired,
  level: oneOf([1, 2, 3, 4, 5, 6]).isRequired,
  img: object,
  children: node,
  className: string,
};

const defaultProps = {
  img: null,
  children: null,
  className: '',
};

/* eslint-disable jsx-a11y/alt-text */
const Title = ({ classes, level, img, children, className, ...props }) => {
  if (React.Children.count(children) < 1) {
    return null;
  }
  const HeadingTag = `h${level}`;

  if (img && img.exist) {
    return (
      <HeadingTag {...props}>
        <img {...img.attr} />
        {children}
      </HeadingTag>
    );
  }

  const headingClasses = classnames({
    [classes[HeadingTag]]: HeadingTag,
    [className]: className,
  });

  return (
    <HeadingTag {...props} className={headingClasses}>
      {children}
    </HeadingTag>
  );
};

Title.propTypes = propTypes;
Title.defaultProps = defaultProps;

export default withStyles(titleStyles)(Title);
