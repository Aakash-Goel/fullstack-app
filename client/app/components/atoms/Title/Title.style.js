const titleStyles = () => ({
  h1: {
    fontSize: '54px',
    lineHeight: '1.14',
    letterSpacing: '1px',
  },
  h2: {
    fontSize: '48px',
    lineHeight: '1.18',
    letterSpacing: '0.8px',
  },
  h3: {
    fontSize: '40px',
    lineHeight: '1.22',
    letterSpacing: '0.6px',
  },
  h4: {
    fontSize: '32px',
    lineHeight: '1.26',
    letterSpacing: '0.4px',
  },
  h5: {
    fontSize: '24px',
    lineHeight: '1.42',
  },
  h6: {
    fontSize: '16px',
    lineHeight: '1.50',
  },
});

export default titleStyles;
