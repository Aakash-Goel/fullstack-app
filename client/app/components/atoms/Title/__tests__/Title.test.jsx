import React from 'react';
import TestRenderer from 'react-test-renderer';

import Title from '../Title';

describe('Title atom', () => {
  it('renders null if children is not present', () => {
    const titleWrapper = TestRenderer.create(<Title level={1} />).toJSON();
    expect(titleWrapper).toMatchSnapshot();
  });

  it('renders correctly if children is present', () => {
    const titleWrapper = TestRenderer.create(
      <Title level={1}>Heading Tag</Title>
    ).toJSON();
    expect(titleWrapper).toMatchSnapshot();
  });

  it('renders correctly with img object', () => {
    const imgProps = {
      exist: true,
      attr: {
        alt: 'Test img',
      },
    };
    const titleWrapper = TestRenderer.create(
      <Title level={2} img={imgProps}>
        Heading with Image tag
      </Title>
    ).toJSON();
    expect(titleWrapper).toMatchSnapshot();
  });
});
