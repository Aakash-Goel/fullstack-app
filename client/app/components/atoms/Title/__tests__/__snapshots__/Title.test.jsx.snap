// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title atom renders correctly if children is present 1`] = `
<h1
  className="Title-h1-1"
>
  Heading Tag
</h1>
`;

exports[`Title atom renders correctly with img object 1`] = `
<h2>
  <img
    alt="Test img"
  />
  Heading with Image tag
</h2>
`;

exports[`Title atom renders null if children is not present 1`] = `null`;
